SasquatchMac/SasquatchMac/ViewControllers/TransmissionViewController.swift

```diff
@@ -0,0 +1,424 @@
+  }
+
+  //Transmission target section
+  private var transmissionTargetSections: [TransmissionTargetSection]?
+  private let kEnabledCellRowIndex = 0
+  private let kAnalyticsCellRowIndex = 1
+  private let kTokenCellRowIndex = 2
+  private let kPauseCellRowIndex = 3
+
+  private class TransmissionTargetSection: NSObject {
+    static var defaultTransmissionTargetIsEnabled: Bool?
+
+    var token: String?
+    var isDefault = false
+
+    func isTransmissionTargetEnabled() -> Bool {
+      if isDefault {
+        return TransmissionTargets.shared.defaultTransmissionTargetIsEnabled
+      } else {
+        return getTransmissionTarget()!.isEnabled()
+      }
+    }
+
+    func setTransmissionTargetEnabled(_ enabledState: Bool) {
+      if !isDefault {
+        getTransmissionTarget()!.setEnabled(enabledState)
+      }
+    }
+
+    func getTransmissionTarget() -> MSAnalyticsTransmissionTarget? {
+      if isDefault {
+        return nil
+      } else {
+        return TransmissionTargets.shared.transmissionTargets[token!]
+      }
+    }
+
+    func shouldSendAnalytics() -> Bool {
+      if isDefault {
+        return TransmissionTargets.shared.defaultTargetShouldSendAnalyticsEvents()
+      } else {
+        return TransmissionTargets.shared.targetShouldSendAnalyticsEvents(targetToken: token!)
+      }
+    }
+
+    func setShouldSendAnalytics(enabledState: Bool) {
+      if isDefault {
+        TransmissionTargets.shared.setShouldDefaultTargetSendAnalyticsEvents(enabledState: enabledState)
+      } else {
+        TransmissionTargets.shared.setShouldSendAnalyticsEvents(targetToken: token!, enabledState: enabledState)
+      }
+    }
+ 
+    func pause() {
+      getTransmissionTarget()!.pause()
+    }
+
+    func resume() {
+      getTransmissionTarget()!.resume()
+    }
+  }
+
+  //Target properties section
+  class EventProperty : NSObject {
+    var key: String = ""
+    var type: String = EventPropertyType.string.rawValue
+    var string: String = ""
+    var double: NSNumber = 0
+    var long: NSNumber = 0
+    var boolean: Bool = false
+    var dateTime: Date = Date.init()
+  }
+
+  enum EventPropertyType : String {
+    case string = "String"
+    case double = "Double"
+    case long = "Long"
+    case boolean = "Boolean"
+    case dateTime = "DateTime"
+
+    static let allValues = [string, double, long, boolean, dateTime]
+  }
+  dynamic var eventProperties = [EventProperty]()
+
+  //Common schema properties section
+  let kDeviceIdRow = 0
+  let kAppNameRow = 1
+  let kAppVersionRow = 2
+  let kAppLocaleRow = 3
+  let kUserIdRow = 4
+  let propertyKeys = ["Collect Device ID", "App Name", "App Version", "App Locale", "User Id"]
+  var propertyValues: [String: [String]]!
+  var collectDeviceIdStates: [String: Bool]!
+  enum CommonSchemaPropertyRow : Int {
+    case appName
+    case appVersion
+    case appLocale
+    case userId
+  }
+
+  public var transmissionTargetMapping: [String]?
+  @IBOutlet weak var defaultTable: NSTableView!
```

Comment:
More class variables - these should all be at the top

Reply:
Resolved.